lualib/resty/kong/log.lua

```diff
@@ -0,0 +1,85 @@
+-- Copyright 2019-2022 Kong Inc.
+
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+
+--    http://www.apache.org/licenses/LICENSE-2.0
+
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+
+local _M = {}
+
+
+local ffi = require("ffi")
+local base = require("resty.core.base")
+base.allows_subsystem("http")
+
+
+ffi.cdef([[
+int ngx_http_lua_kong_ffi_set_log_level(ngx_http_request_t *r, int level);
+]])
+
+
+local get_phase = ngx.get_phase
+local type = type
+local error = error
+local tostring = tostring
+local C = ffi.C
+local get_request = base.get_request
+
+
+local NGX_OK = ngx.OK
+
+
+do
+    local LEVELS = {
+        debug = ngx.DEBUG,
+        info = ngx.INFO,
+        notice = ngx.NOTICE,
+        warn = ngx.WARN,
+        err = ngx.ERR,
+        crit = ngx.CRIT,
+        alert = ngx.ALERT,
+        emerg = ngx.EMERG,
+    }
+
+    local ALLOWED_PHASES = {
+        ["balancer"] = true,
+        ["access"] = true,
+        ["content"] = true,
+    }
+
+    function _M.set_log_level(level)
```

Comment:
WDYT about accepting a log level integer in addition to (or instead of) a string? This way, `ngx` log constants could be used, like so:

```lua
log.set_log_level(ngx.DEBUG)
log.set_log_level(ngx.INFO)
log.set_log_level(ngx.NOTICE)
```